project-geometry/src/main/java/com/esri/arcgismaps/sample/projectgeometry/MainActivity.kt

```diff
@@ -0,0 +1,158 @@
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.symbology.PictureMarkerSymbol
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.mapping.view.MapView
+import com.esri.arcgismaps.sample.projectgeometry.databinding.ActivityMainBinding
+import com.google.android.material.snackbar.Snackbar
+import kotlinx.coroutines.launch
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG = MainActivity::class.java.simpleName
+
+    // set up data binding for the activity
+    private val activityMainBinding: ActivityMainBinding by lazy {
+        DataBindingUtil.setContentView(this, R.layout.activity_main)
+    }
+
+    // setup the data binding for the MapView
+    private val mapView: MapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    // shows the projection information as a TextView
+    private val infoTextView: TextView by lazy {
+        activityMainBinding.infoTextView
+    }
+
+    // setup the red pin marker image as bitmap drawable
+    private val markerDrawable: BitmapDrawable by lazy {
+        // load the bitmap from resources and create a drawable
+        val bitmap = BitmapFactory.decodeResource(resources, R.drawable.pin_symbol)
+        BitmapDrawable(resources, bitmap)
+    }
+
+    // setup the red pin marker as a Graphic
+    private val markerGraphic: Graphic by lazy {
+        // creates a symbol from the marker drawable
+        Graphic().apply {
+            symbol = PictureMarkerSymbol(markerDrawable).apply {
+                // resize the symbol into a smaller size
+                width = 30f
+                height = 30f
+                // offset in +y axis so the marker spawned
+                // is right on the touch point
+                offsetY = 25f
+            }
+        }
+    }
+
+    // creates a graphic overlay
+    private val graphicsOverlay: GraphicsOverlay = GraphicsOverlay()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // configure mapView assignments
+        mapView.apply {
+            // create and add a map with a navigation night basemap style
+            map = ArcGISMap(BasemapStyle.ArcGISNavigationNight)
+            // add the graphics overlay to display marker graphics
+            graphicsOverlays.add(graphicsOverlay)
+            // set the default viewpoint to Redlands,CA
+            setViewpoint(Viewpoint(34.058, -117.195, 5e4))
+        }
+
+        lifecycleScope.launch {
+            // check if the map has loaded successfully
+            mapView.map?.load()?.onSuccess {
```

Comment:
These nullability `?` operators can be avoided by storing the map as a local variable above
```kotlin
// (above...)
val map = ArcGISMap(BasemapStyle.ArcGISNavigationNight)
mapView.apply {
    map = map
}

// (here...)
map.load().onSuccess { // ...
```